graylog2-server/src/main/java/org/graylog2/rest/resources/streams/destinations/filters/StreamDestinationFiltersResource.java

```diff
@@ -0,0 +1,187 @@
+                              @ApiParam(name = "filterId", required = true) @PathParam("filterId") @NotBlank String filterId) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkPermission(RestPermissions.STREAM_DESTINATION_FILTERS_READ, filterId);
+        checkStream(streamId);
+
+        final var dto = filterService.findById(filterId)
+                .orElseThrow(() -> new NotFoundException("Filter not found"));
+
+        return Response.ok(wrapDto(dto)).build();
+    }
+
+    @POST
+    @Path("/filters")
+    @ApiOperation("Create new filter rule")
+    @AuditEvent(type = AuditEventTypes.STREAM_DESTINATION_FILTER_CREATE)
+    public Response createFilter(@ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+                                 @ApiParam(name = "JSON Body", required = true) @Valid StreamDestinationFilterRuleDTO dto) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkPermission(RestPermissions.STREAM_DESTINATION_FILTERS_CREATE);
+        checkStream(streamId);
+
+        return Response.ok(wrapDto(filterService.create(streamId, dto))).build();
+    }
+
+    @PUT
+    @Path("/filters/{filterId}")
+    @ApiOperation(value = "Update filter rule")
+    @AuditEvent(type = AuditEventTypes.STREAM_DESTINATION_FILTER_UPDATE)
+    public Response updateFilter(@ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+                                 @ApiParam(name = "filterId", required = true) @PathParam("filterId") @NotBlank String filterId,
+                                 @ApiParam(name = "JSON Body", required = true) @Valid StreamDestinationFilterRuleDTO dto) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkPermission(RestPermissions.STREAM_DESTINATION_FILTERS_EDIT, filterId);
+        checkStream(streamId);
+
+        if (!filterId.equals(dto.id())) {
+            throw new BadRequestException("The filter ID in the URL doesn't match the one in the payload");
+        }
+
+        return Response.ok(wrapDto(filterService.update(streamId, dto))).build();
+    }
+
+    @DELETE
+    @Path("/filters/{filterId}")
+    @ApiOperation("Delete filter rule")
+    @AuditEvent(type = AuditEventTypes.STREAM_DESTINATION_FILTER_DELETE)
+    public Response deleteFilter(@ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+                                 @ApiParam(name = "filterId", required = true) @PathParam("filterId") @NotBlank String filterId) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkPermission(RestPermissions.STREAM_DESTINATION_FILTERS_DELETE, filterId);
+        checkStream(streamId);
+
+        return Response.ok(wrapDto(filterService.delete(filterId))).build();
```

Comment:
possible to delete a filter that is not associated with the streamId

Reply:
Aye, same as for the `find*` methods. We should scope the delete to the `streamId`.

Reply:
Done in https://github.com/Graylog2/graylog2-server/pull/19739/commits/28e188f0cf88223ea8e581884e5606be6defbba9.